sigal/plugins/compress_assets.py

```diff
@@ -0,0 +1,156 @@
+"""Plugin to compress static files for faster HTTP transfer.
+
+Currently, 3 methods are supported:
+    - gzip: No dependency required.
+            This is the fastest, but also largest output.
+    - zopfli: Need zopfli module from https://pypi.python.org/pypi/zopfli.
+              gzip compatible output with optimized size.
+    - brotli: Need brotli module from https://pypi.python.org/pypi/Brotli.
+              Brotli is the best compressor for web usage.
+
+"""
+
+import logging
+import gzip
+import shutil
+import os
+
+from sigal import signals
+from click import progressbar
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_SETTINGS = {
+        'suffixes': ['htm', 'html', 'css', 'js', 'svg'],
+        'method': 'gzip',
+        }
+
+
+class BaseCompressor:
+    suffix = None
+
+    def __init__(self, settings):
+        self.suffixes_to_compress = settings.get('suffixes', DEFAULT_SETTINGS['suffixes'])
+
+    def do_compress(self, filename, compressed_filename):
+        '''
+        Perform actual compression.
+        This should be implemented by subclasses.
+        '''
+        raise NotImplementedError
+
+    def compress(self, filename):
+        '''
+        Compress a file, only if needed.
+        '''
+        compressed_filename = self.get_compressed_filename(filename)
+        if not compressed_filename:
+            return
+
+        self.do_compress(filename, compressed_filename)
+
+    def get_compressed_filename(self, filename):
+        '''
+        If the given filename should be compressed, returns the compressed filename.
+        A file can be compressed if:
+            - It is a whitelisted extension
+            - The compressed file does not exist
+            - The compressed file exists by is older than the file itself
+        Otherwise, it returns False.
+        '''
+        if not os.path.splitext(filename)[1][1:] in self.suffixes_to_compress:
```

Comment:
This could fail is a file has no extension ? (`splitext` returns an empty string, so slicing with `[1:]` will fail I guess)

Reply:
```
>>> os.path.splitext('test')[1][1:]
''
```
In the beginning, I thought you were right. The slicing does not failed. This is also a surprise to me 😄

Reply:
Oh, surprising !